aptos-move/aptos-vm/src/move_vm_ext/session.rs

```diff
@@ -428,14 +428,17 @@ impl<'r> WriteOpConverter<'r> {
         use MoveStorageOp::*;
         use WriteOp::*;
 
-        let existing_value_opt = self.remote.get_state_value(state_key).map_err(|_| {
-            VMStatus::error(
-                StatusCode::STORAGE_ERROR,
-                err_msg("Storage read failed when converting change set."),
-            )
-        })?;
-
-        let write_op = match (existing_value_opt, move_storage_op) {
+        let maybe_existing_metadata =
```

Comment:
Instead of using existing value, we are now using the metadata of existing value. What's the motivation behind this change?

Reply:
We never needed the value there - only metadata to calculate storage refunds. Motivation behind having a resolver is to avoid using a fixed `get_state_value` method and use resolver abstraction so we can use any storage backend - for indexer, execution, etc.

Maybe we should expose this in resource/move resolvers, but this suffices for now I think